pkg/client/subjectaccessreview.go

```diff
@@ -21,40 +33,51 @@ type SubjectAccessReviewInterface interface {
 
 // subjectAccessReviews implements SubjectAccessReviewsNamespacer interface
 type subjectAccessReviews struct {
-	r  *Client
-	ns string
+	r     *Client
+	ns    string
+	token string
 }
 
 // newSubjectAccessReviews returns a subjectAccessReviews
-func newSubjectAccessReviews(c *Client, namespace string) *subjectAccessReviews {
+func newSubjectAccessReviews(c *Client, namespace, token string) *subjectAccessReviews {
 	return &subjectAccessReviews{
-		r:  c,
-		ns: namespace,
+		r:     c,
+		ns:    namespace,
+		token: token,
 	}
 }
 
 // Create creates new policy. Returns the server's representation of the policy and error if one occurs.
 func (c *subjectAccessReviews) Create(policy *authorizationapi.SubjectAccessReview) (result *authorizationapi.SubjectAccessReviewResponse, err error) {
 	result = &authorizationapi.SubjectAccessReviewResponse{}
-	err = c.r.Post().Namespace(c.ns).Resource("subjectAccessReviews").Body(policy).Do().Into(result)
+	err = overrideAuth(c.token, c.r.Post().Namespace(c.ns).Resource("subjectAccessReviews")).Body(policy).Do().Into(result)
 	return
 }
 
 // clusterSubjectAccessReviews implements ClusterSubjectAccessReviews interface
 type clusterSubjectAccessReviews struct {
-	r *Client
+	r     *Client
+	token string
 }
 
 // newClusterSubjectAccessReviews returns a clusterSubjectAccessReviews
-func newClusterSubjectAccessReviews(c *Client) *clusterSubjectAccessReviews {
+func newClusterSubjectAccessReviews(c *Client, token string) *clusterSubjectAccessReviews {
 	return &clusterSubjectAccessReviews{
-		r: c,
+		r:     c,
+		token: token,
 	}
 }
 
 // Create creates new policy. Returns the server's representation of the policy and error if one occurs.
 func (c *clusterSubjectAccessReviews) Create(policy *authorizationapi.SubjectAccessReview) (result *authorizationapi.SubjectAccessReviewResponse, err error) {
 	result = &authorizationapi.SubjectAccessReviewResponse{}
-	err = c.r.Post().Resource("subjectAccessReviews").Body(policy).Do().Into(result)
+	err = overrideAuth(c.token, c.r.Post().Resource("subjectAccessReviews")).Body(policy).Do().Into(result)
 	return
 }
+
+func overrideAuth(token string, req *client.Request) *client.Request {
```

Comment:
this gets weird if the original client was using cert auth... is there no way to avoid leaking the descriptor? this feels like an odd way to fix the issue